### PR TITLE
Remove default methods using lazy fallbacks

### PR DESCRIPTION
**Pull request by romainfrancois**
Part of #5573. We can later remove the functions.

### CODE REVIEW

File: R/group-by.r [hunk 1]
```diff
@@ -144,7 +144,7 @@ group_by_prepare <- function(.data, ..., .add = FALSE, .dots = deprecated(), add
   new_groups <- enquos(...)
   if (!missing(.dots)) {
     # Used by dbplyr 1.4.2 so can't aggressively deprecate
-    lifecycle::deprecate_soft("1.0.0", "group_by(.dots = )")
+    lifecycle::deprecate_warn("1.0.0", "group_by(.dots = )")
     new_groups <- c(new_groups, compat_lazy_dots(.dots, env = caller_env()))
```
Comment (romainfrancois) [inline R/group-by.r:148]: This is the only use of `compat_lazy_dots()` outside of `deprec-lazyeval.R`

**Comment by romainfrancois** [comment]
closes #5083

**Comment by hadley** [comment]
Can you do a revdep check run to see if this breaks any CRAN packages?

**Comment by hadley** [comment]
(Otherwise looks good, just needs a news bullet, saying something like: Removed default fallbacks to lazyeval methods; this will yield better error messages when you call a dplyr function with the wrong input, and is part of our long term plan to remove the deprecate lazyeval interface).

**Comment by romainfrancois** [comment]
This breaks: 

- AeRobiology because of `ggvis` (`filter()`) 
- `ggvis` 

- `lplyr` : `mutate()`
- `plotly` : `mutate()`

- `timetk`: not sure why
- `TextMiningGUI`: probably unrelated

**Comment by romainfrancois** [comment]
```` 
# AeRobiology

<details>

* Version: 2.0.1
* GitHub: NA
* Source code: https://github.com/cran/AeRobiology
* Date/Publication: 2019-06-03 06:20:03 UTC
* Number of recursive dependencies: 95

Run `cloud_details(, "AeRobiology")` for more info

</details>

## Newly broken

*   checking examples ... ERROR
    ```
    Running examples in ‘AeRobiology-Ex.R’ failed
    The error most likely occurred in:
    
    > ### Name: iplot_pollen
    > ### Title: Interactive Plotting Pollen Data (one season).
    > ### Aliases: iplot_pollen
    > 
    > ### ** Examples
    > 
    > data("munich_pollen")
    > iplot_pollen(data = munich_pollen, year = 2012)
    Error in UseMethod("filter") : 
      no applicable method for 'filter' applied to an object of class "ggvis"
    Calls: iplot_pollen ... eval -> eval -> _fseq -> freduce -> <Anonymous> -> filter
    Execution halted
    ```

# ggvis

<details>

* Version: 0.4.6
* GitHub: NA
* Source code: https://github.com/cran/ggvis
* Date/Publication: 2020-10-02 00:10:02 UTC
* Number of recursive dependencies: 67

Run `cloud_details(, "ggvis")` for more info

</details>

## Newly broken

*   checking examples ... ERROR
    ```
    Running examples in ‘ggvis-Ex.R’ failed
    The error most likely occurred in:
    
    > ### Name: auto_group
    > ### Title: Automatically group data by grouping variables
    > ### Aliases: auto_group
    > 
    > ### ** Examples
    > 
    > # One line
    > mtcars %>% ggvis(~disp, ~mpg, stroke = ~factor(cyl)) %>% layer_paths()
    > # One line for each level of cyl
    > mtcars %>% ggvis(~disp, ~mpg, stroke = ~factor(cyl)) %>% group_by(cyl) %>%
    +   layer_paths()
    Error in UseMethod("group_by") : 
      no applicable method for 'group_by' applied to an object of class "ggvis"
    Calls: %>% ... eval -> _fseq -> freduce -> <Anonymous> -> group_by
    Execution halted
    ```

*   checking tests ... ERROR
    ```
      Running ‘testthat.R’
    Running the tests in ‘tests/testthat.R’ failed.
    Last 13 lines of output:
        8.         └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
        9.           └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
       10.             └─`_fseq`(`_lhs`)
       11.               └─magrittr::freduce(value, `_function_list`)
       12.                 └─function_list[[i]](value)
       13.                   └─dplyr::group_by(., cyl)
      
      ══ testthat results  ═══════════════════════════════════════════════════════════
      Warning (test-compute-stack.r:11:3): compute_stack works as expected
      Warning (test-compute-stack.r:11:3): compute_stack works as expected
      ERROR (test-specs.r:5:3): (code run outside of `test_that()`)
      
      [ FAIL 1 | WARN 2 | SKIP 0 | PASS 443 ]
      Error: Test failures
      Execution halted
    ```

# lplyr

<details>

* Version: 0.1.12
* GitHub: https://github.com/paulponcet/lplyr
* Source code: https://github.com/cran/lplyr
* Date/Publication: 2017-11-05 04:05:00 UTC
* Number of recursive dependencies: 60

Run `cloud_details(, "lplyr")` for more info

</details>

## Newly broken

*   checking examples ... ERROR
    ```
    Running examples in ‘lplyr-Ex.R’ failed
    The error most likely occurred in:
    
    > ### Name: mutate_.list
    > ### Title: Dplyr verbs for lists and pairlists
    > ### Aliases: mutate_.list mutate_.pairlist rename_.list rename_.pairlist
    > 
    > ### ** Examples
    > 
    > xs <- list(x1 = 1:3, 
    +            x2 = 2:5, 
    +            x3 = list("alpha", c("beta", "gamma")))
    > 
    > # Non-standard evaluation
    > mutate(xs, x4 = 4)
    Error in UseMethod("mutate") : 
      no applicable method for 'mutate' applied to an object of class "list"
    Calls: mutate
    Execution halted
    ```

# plotly

<details>

* Version: 4.9.2.1
* GitHub: https://github.com/ropensci/plotly
* Source code: https://github.com/cran/plotly
* Date/Publication: 2020-04-04 19:50:02 UTC
* Number of recursive dependencies: 152

Run `cloud_details(, "plotly")` for more info

</details>

## Newly broken

*   checking examples ... ERROR
    ```
    ...
     2 Abilene  2000     2    98  6505000  58700      746       6.6 2000.
     3 Abilene  2000     3   130  9285000  58100      784       6.8 2000.
     4 Abilene  2000     4    98  9730000  68600      785       6.9 2000.
     5 Abilene  2000     5   141 10590000  67300      794       6.8 2000.
     6 Abilene  2000     6   156 13910000  66900      780       6.6 2000.
     7 Abilene  2000     7   152 12635000  73500      742       6.2 2000.
     8 Abilene  2000     8   131 10710000  75000      765       6.4 2001.
     9 Abilene  2000     9   104  7615000  64500      771       6.5 2001.
    10 Abilene  2000    10   101  7040000  59300      764       6.6 2001.
    # … with 8,592 more rows
    > 
    > # dplyr verbs operate on plotly objects as if they were data frames
    > p <- economics %>%
    +   plot_ly(x = ~date, y = ~unemploy / pop) %>%
    +   add_lines() %>%
    +   mutate(rate = unemploy / pop) %>% 
    +   filter(rate == max(rate))
    Error in UseMethod("mutate") : 
      no applicable method for 'mutate' applied to an object of class "c('plotly', 'htmlwidget')"
    Calls: %>% ... eval -> eval -> _fseq -> freduce -> <Anonymous> -> mutate
    Execution halted
    ```

*   checking tests ... ERROR
    ```
      Running ‘testthat.R’
    Running the tests in ‘tests/testthat.R’ failed.
    Last 13 lines of output:
      Warning (test-plotly-subplot.R:396:3): shape xref/yref bumping
      Warning (test-plotly-subplot.R:396:3): shape xref/yref bumping
      Warning (???): May specify legendgroup with through a vector of values
      Warning (test-plotly-symbol.R:30:3): Setting a constant symbol works
      Warning (test-plotly-symbol.R:68:3): Trace ordering is alphabetical
      Warning (test-plotly-symbol.R:68:3): Trace ordering is alphabetical
      Warning (test-plotly.R:41:3): Variable mappings return same result regardless of where they appear
      Warning (test-plotly.R:41:3): Variable mappings return same result regardless of where they appear
      Warning (test-plotly.R:143:3): Character strings correctly mapped to a positional axis
      Warning (test-plotly.R:143:3): Character strings correctly mapped to a positional axis
      ERROR (test-plotly.R:179:3): Complex example works
      
      [ FAIL 3 | WARN 65 | SKIP 54 | PASS 1363 ]
      Error: Test failures
      Execution halted
    ```

## In both

*   checking installed package size ... NOTE
    ```
      installed size is  6.8Mb
      sub-directories of 1Mb or more:
        htmlwidgets   3.7Mb
    ```

# TextMiningGUI

<details>

* Version: 0.1
* GitHub: NA
* Source code: https://github.com/cran/TextMiningGUI
* Date/Publication: 2020-08-11 15:20:08 UTC
* Number of recursive dependencies: 158

Run `cloud_details(, "TextMiningGUI")` for more info

</details>

## Newly broken

*   checking examples ... ERROR
    ```
    Running examples in ‘TextMiningGUI-Ex.R’ failed
    The error most likely occurred in:
    
    > ### Name: TextMiningGUI
    > ### Title: TextMiningGUI
    > ### Aliases: TextMiningGUI
    > 
    > ### ** Examples
    > 
    > library(TextMiningGUI)
    > if(TextMiningGUI()){}
    Error in structure(.External(.C_dotTclObjv, objv), class = "tclObj") : 
      [tcl] invalid command name "toplevel".
    Calls: TextMiningGUI ... tktoplevel -> tkwidget -> tcl -> .Tcl.objv -> structure
    Execution halted
    ```

*   checking whether package ‘TextMiningGUI’ can be installed ... WARNING
    ```
    Found the following significant warnings:
      Warning: loading Rplot failed
    See ‘/tmp/workdir/TextMiningGUI/new/TextMiningGUI.Rcheck/00install.out’ for details.
    ```

# timetk

<details>

* Version: 2.5.0
* GitHub: https://github.com/business-science/timetk
* Source code: https://github.com/cran/timetk
* Date/Publication: 2020-10-22 15:40:02 UTC
* Number of recursive dependencies: 189

Run `cloud_details(, "timetk")` for more info

</details>

## Newly broken

*   checking examples ... ERROR
    ```
    ...
     7 D10   2014-07-09 2019.
     8 D10   2014-07-10 2007.
     9 D10   2014-07-11 2010 
    10 D10   2014-07-12 2002.
    # … with 10,133 more rows
    > 
    > # Remove Non-Working Days (Weekends & Holidays)
    > holidays <- tk_make_holiday_sequence(
    +     start_date = "2017-01-01",
    +     end_date   = "2017-12-31",
    +     calendar   = "NYSE")
    > 
    > FANG %>%
    +     group_by(symbol) %>%
    +     future_frame(.
    length_out       = "1 year",
    +                  .inspect_weekdays = TRUE,
    +                  .skip_values      = holidays)
    .date_var is missing. Using: date
    Error in is_quosure(x) : argument "x" is missing, with no default
    Calls: %>% ... switch_expr -> quo_squash_impl -> switch_expr -> is_quosure
    Execution halted
    ```

## In both

*   checking dependencies in R code ... NOTE
    ```
    Namespace in Imports field not imported from: ‘generics’
      All declared Imports should be used.
    ```

*   checking data for non-ASCII characters ... NOTE
    ```
      Note: found 2750 marked UTF-8 strings
    ```

````

**Comment by hadley** [comment]
This feels worthwhile to me. Maybe I can do PRs?

**Comment by hadley** [comment]
* AeRobiology: need to check ggvis update fixes
* ggvis: https://github.com/rstudio/ggvis/pull/488
* lplyr:: emailed to confirm that it's worth updating
* timetk: I see the problem with CRAN dplyr (https://github.com/business-science/timetk/issues/66)
* plotly: already fixed in master

**Pull request merged**